src/Distributed/multi_architectures.jl

```diff
@@ -103,24 +96,26 @@ function DistributedArch(child_architecture = CPU();
         isnothing(devices) ? device!(node_rank % ndevices()) : device!(devices[node_rank+1]) 
     end
 
-    B = use_buffers
+    mpi_requests = enable_overlapped_computation ? MPI.Request[] : nothing
 
-    return DistributedArch{A, R, I, ρ, C, γ, B}(child_architecture, local_rank, local_index, ranks, local_connectivity, communicator)
-end
+    M = typeof(mpi_requests)
+    T = typeof(Ref(0))
 
-const ViewsDistributedArch = DistributedArch{<:Any, <:Any, <:Any, <:Any, <:Any, <:Any, false}
+    return DistributedArch{A, R, I, ρ, C, γ, M, T}(child_architecture, local_rank, local_index, ranks, local_connectivity, communicator, mpi_requests, Ref(0))
+end
 
-using_buffered_communication(::DistributedArch{A, R, I, ρ, C, γ, B}) where {A, R, I, ρ, C, γ, B} = B
+const BlockingDistributedArch = DistributedArch{<:Any, <:Any, <:Any, <:Any, <:Any, <:Any, <:Nothing}
```

Comment:
```suggestion
const BlockingDistributedArch = DistributedArch{<:Nothing}
```

Why is the dispatched type parameter so far down the list? It's easier to read the code if the dispatched parameter is moved to the front as in the above.

Reply:
Move it to the front